### ISSUE TITLE
Accessibility: Improving ipython news page

### ISSUE THREAD

**Issue by ChristianaOsun** [jupyterhub/outreachy#119]
### What page is this for?

https://ipython.org/news.html

### WAVE accessibility report

The WAVE accessibility tool showed that the following can be improved upon on the [ipython news](https://ipython.org/news.html) page.

![WAVE](https://user-images.githubusercontent.com/66247893/197401080-3bba2988-89c1-4596-b4a4-c351344f65ef.png)

Based on this report, the following can be improved upon using the WCAG guidelines:

1. *Language missing or invalid:* The <html lang> attribute is missing. This should be added as it helps to facilitate the translation of the document. 
2. *Contrast Error:* The <href> html element on the documents of the page have very low contrast. The contrast between the text and background is very low and makes the page not very accessible for readers on the low visibility spectrum.
This can be improved by increasing the contrast between the text color and the background color. According to WCAG, a text contrast ratio of at least 4.5:1 is recommended, and ideal.
The <href> html element on the documents of the page have very low contrast. The contrast ratio for the <div class="rel"> is currently at 4.07:1. I suggest that this should be increased.

3. *Alternate text:* There are 4 suspicious alt attributes tag as they were not properly named. When visually impaired readers visit the page or when the page does not load properly, the alt tag does not convey adequate information about the page.

### The HTML element to be changed

```html
1. Language attribute: Include <html lang="en"> just before the <head> section which contains information about the page.

Code: 
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<base href="https://ipython.org/news.html">
<meta http-equiv="X-UA-Compatible" content="IE=Edge">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">

2. Contrast Error: The <href> html element on the documents of the page have very low contrast.
<div class="rel">
<a href="https://jupyter.readthedocs.io/en/latest/install.html" target="_blank" style="color: rgb(206, 92, 0); background-color: rgb(255, 255, 255);">
Install
</a>

<a href="donate.html" style="color: rgb(206, 92, 0); background-color: rgb(255, 255, 255);">
Donate
</a>

div.header div.rel a {
  color: #ce5c00;
}
---
<a class="headerlink" href="#news" title="Permalink to this headline" style="color: rgb(221, 221, 221); background-color: rgb(255, 255, 255);">
</a>
-----
a {
  color: #ce5c00;
}
-----
a.headerlink {
  visibility: hidden;
  color: #dddddd;
  padding-left: .3em;
}
----
h1, h2, h3, h4 {
  font-weight: normal;
  color: #3465a4;
  margin-bottom: .8em;
}
3. Missing/suspicious alternate text: 
<a href="index.html">
<img class="logo" src="_static/IPy_header.png" alt="Logo">
</a></div>
---
<a class="reference external" href="https://ipython.org/ipython-doc/rel-0.11/whatsnew/version0.11.html">
----
<img alt="_images/ipy_0.11.png" src="_images/ipy_0.11.png">

4. Image with title attribute: 
<a href="http://nbviewer.ipython.org">
<img src="_static/nbviewer-thm-4.png" title="IPython Notebook Viewer">
</a>
---
<img title="Packt Publishing donates a portion of the proceeds from this book to support IPython's development." width="160" height="200" id="sidebar-book-cover" style="margin-bottom: 10px;" src="_static/jupyter-notebook-for-all-I.jpg">

5. Suspicious Link Text:
<p>
On October 11, 2010, we released IPython 0.10.1. The full release notes can be found
<a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/html/changes.html#release-0-10-1">
here
</a>
, describing in detail the changes in this release.
---
<a class="reference external" href="http://channel9.msdn.com/shows/The+HPC+Show/Open-source-HPC-code-Episode-12-IPython-computes-150-million-digits-of-Pi-in-Parallel/">
videos
</a>
about using IPython on Windows HPC Server 2008. Additional information about these features can be found in our documentation
<a class="reference external" href="http://ipython.scipy.org/doc/nightly/html/parallel/parallel_winhpc.html">
here
</a>
</p>
---
<p>
On August 4 2009, we’ve released version 0.10 of IPython. The full release notes can be found
<a class="reference external" href="http://ipython.scipy.org/doc/rel-0.10/html/changes.html#release-0-10">
here
</a>
, describing in detail all new features, bug fixes and API changes of this release.
</p>
---
<p>
On September 14, 2008, the IPython team is proud to release version 0.9 final. See
<a class="reference external" href="http://ipython.scipy.org/announcements/ann-ipython-0.9.txt">
here
</a>
for the full announcement; a detailed description of the changes can be found
<a class="reference external" href="http://ipython.scipy.org/doc/rel-0.9/html/changes.html#release-0-9">
here
</a>
.
</p>
---
<p>
<a class="reference external" href="http://wiki.python.org/moin/IPyKit">
IpyKit
</a>

is a standalone IPython executable, created with py2exe and as such works on machines without python (or where you just don’t have time to mess with installation). It includes pyreadline for full tab completion and color support. Get the first release (fully functional, based on svn IPython)
<a class="reference external" href="http://vivainio.googlepages.com/ipykit.zip">
here
</a>

. Just unzip and run ipython.exe to go.
</p>
---
<p>
On April 28, we’ll hold a coding sprint to push the saw branch and the integration of the trunk into it. Details
<a class="reference external" href="http://ipython.scipy.org/moin/Developer_Zone/Sprint">
here
</a>
.
</p>
----
<p>
IPython 0.7.3 can be downloaded
<a class="reference external" href="http://ipython.scipy.org/dist">
here
</a>
.
</p>
----
<p>
IPython 0.7.2 can be downloaded
<a class="reference external" href="https://ipython.org/download.html">
here
</a>
.
</p>

6. Images with alternative text:
<a href="index.html">
<img class="logo" src="_static/IPy_header.png" alt="Logo">
</a></div>
-----
<a href="http://jupytercon.com">
<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2017">
</a>

7. Aria Label:
<div class="sphinxsidebar" role="navigation" aria-label="main navigation">
```

### Your proposed HTML (or CSS), after change

```html
1. Language attribute:
<html xmlns="http://www.w3.org/1999/xhtml">
<html lang="en">
<head>
<base href="https://ipython.org/news.html">
<meta http-equiv="X-UA-Compatible" content="IE=Edge">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">

2. Contrast Errors:
<div class="rel">
<a href="https://jupyter.readthedocs.io/en/latest/install.html" target="_blank" style="color: rgb(122, 55, 0); background-color: rgb(250, 250, 250);">
Install
</a>

<a href="donate.html" style="color: rgb(133, 60, 0); background-color: rgb(255, 255, 255);">
Donate
</a>

div.header div.rel a {
  color: #853c00;
}
------
<a class="headerlink" href="#news" title="Permalink to this headline" style="color: rgb(87, 87, 87); background-color: rgb(255, 255, 255);">
</a>
----
a {
  color: #853c00;
}
-----
a.headerlink {
  visibility: hidden;
  color: #575757;
  padding-left: .3em;
}
-----
h1, h2, h3, h4 {
  font-weight: normal;
  color: #2C558C;
  margin-bottom: .8em;
}

3. Missing/Suspicious alternate text: 
<a href="index.html">
<img class="logo" src="_static/IPy_header.png" alt="ipython logo">
</a></div>
---
<a class="reference external" alt="ipython NewLook" href="https://ipython.org/ipython-doc/rel-0.11/whatsnew/version0.11.html">
----
<img alt="ipython enhanced look" src="_images/ipy_0.11.png">

4. Image with title attribute: 
<a href="http://nbviewer.ipython.org">
<img src="_static/nbviewer-thm-4.png" title="IPython Notebook Viewer" alt="Notebook Viewer">
</a>
---
<img title="Packt Publishing donates a portion of the proceeds from this book to support IPython's development." width="160" height="200" id="sidebar-book-cover" style="margin-bottom: 10px;" alt="Jupyter Notebook Coverpage" src="_static/jupyter-notebook-for-all-I.jpg">

5. Suspicious Link Text:
<p>
On October 11, 2010, we released IPython 0.10.1. The <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/html/changes.html#release-0-10-1">
full release notes
</a>
, describing in detail the changes in this release.
</p>
---
<a class="reference external" href="http://channel9.msdn.com/shows/The+HPC+Show/Open-source-HPC-code-Episode-12-IPython-computes-150-million-digits-of-Pi-in-Parallel/">
videos
</a>
about using IPython on Windows HPC Server 2008. Additional information about these features can be found in our <a class="reference external" href="http://ipython.scipy.org/doc/nightly/html/parallel/parallel_winhpc.html">
documentation
</a>
</p>
---
<p>
On August 4 2009, we’ve released version 0.10 of IPython. The <a class="reference external" href="http://ipython.scipy.org/doc/rel-0.10/html/changes.html#release-0-10">
full release notes
</a>
describing in detail all new features, bug fixes and API changes of this release.
</p>
---
<p>
On September 14, 2008, the IPython team is proud to release
<a class="reference external" href="http://ipython.scipy.org/announcements/ann-ipython-0.9.txt">
version 0.9 final
</a>
. For the full announcement; a
<a class="reference external" href="http://ipython.scipy.org/doc/rel-0.9/html/changes.html#release-0-9">
detailed description
</a>
of the changes can be found here.
</p>
---
<p>
<a class="reference external" href="http://wiki.python.org/moin/IPyKit">
IpyKit
</a>

is a standalone IPython executable, created with py2exe and as such works on machines without python (or where you just don’t have time to mess with installation). It includes pyreadline for full tab completion and color support. Get the
<a class="reference external" href="http://vivainio.googlepages.com/ipykit.zip">
first release
</a>
(fully functional, based on svn IPython). Just unzip and run ipython.exe to go.
</p>
----
<p>
On April 28, we’ll hold a coding sprint to push the saw branch and the integration of the trunk into it. 
<a class="reference external" href="http://ipython.scipy.org/moin/Developer_Zone/Sprint">
Details
</a>
.
</p>
-----
<p>
<a class="reference external" href="http://ipython.scipy.org/dist">
IPython 0.7.3
</a>
can be downloaded.
</p>
----
<p>
<a class="reference external" href="https://ipython.org/download.html">
IPython 0.7.2 </a>
can be downloaded.
</p>

6. Images with alternative text:
<a href="index.html">
<img class="logo" src="_static/IPy_header.png" alt="">
</a></div>

-----
<a href="http://jupytercon.com">
<img src="_static/jupytercon-logo.svg" width="180px" alt="">
</a>

7. 7. Aria Label:
<div class="sphinxsidebar" role="navigation" aria-label="main navigation" label="navigation">
```

### More info

The inline styling seem to conflict with the external styling for some reasons. For best result, I will suggest that all inline styling be removed and only the external stylesheet should be used. This is to avoid errors that might occur as a result of using inline, internal and external stylesheet on the document. Also, because it is difficult to manage and update inline style.

**Comment by minrk** [comment]
Good suggestions, thank you!

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
